small_game_canvas/index.html

```diff
@@ -0,0 +1,130 @@
+<!DOCTYPE HTML/>
+<html>
+
+<head>
+    <meta charset="utf-8" />
+    <meta lang="en-UK" />
+    <title>small_game_online</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+<!--    <script src="app.js" type="text/javascript"></script>-->
+</head>
+
+<body onload="startGame()">
+
+    <script>
+
+        //variables to declare:
+        var symbolImage, gameInformation, symbolList, spinButton, fruitNames, background;
+
+
+        function startGame() {
+
+            //gameArea starts:
+            myGameArea.start();
+
+            //components:
+            images = [
+                new component(235, 155, "strawberry.jpg", 70, 200, "image"),
+                new component(235, 155, "lemon.png", 70, 200, "image"),
+                new component(235, 155, "kiwi.png", 70, 200, "image")
+            ];
+            active = 0;
+
+            gameInformation = new component(235, 155, "gold", 312, 202);
+
+            symbolList = new component(234, 155, "white", 554, 200);
+
+            fruitNames = new component("30px", "serif", "black", 554, 130, "text");
+
+            spinButton = new component(98, 98, "spin_button.png", 824, 220, "image");
+
+            background = new component(960, 536, "bg-ice.jpg", 0, 0, "image");
+
+        }
+
+        //this is what the gameArea does:
+        var myGameArea = {
+
+            //canvas is created:
+            canvas : document.createElement("canvas"),
+            start : function() {
+                this.canvas.width = 960;
+                this.canvas.height = 536;
+                this.context = this.canvas.getContext("2d");
+                document.body.insertBefore(this.canvas, document.body.childNodes[0]);
+                this.interval = setInterval(updateGameArea, 500);
+            },
+
+            clear : function() {
+                this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+
+            }
+
+
+        }
+
+        //this is what happens with components duting the game:
+        function component(width, height, color, x, y, type) {

```

Comment:
Rather than having one component that fits all, it's usually better to create separate components for each type: `imageComponent`, `rectangleComponent` and `textComponent`.